pandas/core/groupby/groupby.py

```diff
@@ -865,28 +882,52 @@ def _wrap_applied_output(self, keys, values, not_indexed_same: bool = False):
     def _cython_agg_general(
         self, how: str, alt=None, numeric_only: bool = True, min_count: int = -1
     ):
-        output = {}
+        output: Dict[int, np.ndarray] = collections.OrderedDict()
```

Comment:
dict is ok now

Reply:
Do we care about non Cpython implementations? I think could break those by making this just a plain dict; not sure we care

Reply:
IIRC there was an issue a few weeks ago about pypy and something like 700 test failures.  we use CPython a lot in our cython code so we're kinda locked in.  That said, reminders that we're relying on implementation details are always welcome

Reply:
maybe but unless we have a pypy build and actually support it
we r not supporting it

you can open an issue for discussion

Reply:
i think pypy dicts are ordered as well

Reply:
OK switched to dict